hledger-lib/Hledger/Data/Amount.hs

```diff
@@ -434,9 +435,9 @@ applyDigitGroupStyle (Just (DigitGroups c gs)) s = addseps (repeatLast gs) s
   where
     addseps [] s = s
     addseps (g:gs) s
-      | length s <= g = s
-      | otherwise     = let (part,rest) = splitAt g s
-                        in part ++ [c] ++ addseps gs rest
+      | genericLength s <= toInteger g = s
```

Comment:
What do you think about `genericLength (length s) < genericLength g` here, as a general pattern to avoid any risk of slowness from genericLength ?

Reply:
I presume you mean `toInteger (length s) < toInteger g`. I concur.

Reply:
Doh. Yes thank you for reading my mind.